sw/device/sca/ecc256_keygen_serial.c

```diff
@@ -0,0 +1,305 @@
+ * @param[in] mask  Mask for seed.
+ * @param[out] d0   First share of masked private key d.
+ * @param[out] d1   Second share of masked private key d.
+ */
+static void p256_ecdsa_gen_secret_key(const uint32_t *seed,
+                                      const uint32_t *mask, uint32_t *d0,
+                                      uint32_t *d1) {
+  // Run the key generation program.
+  p256_run_keygen(kEcc256ModePrivateKeyOnly, seed, mask);
+
+  // Read results.
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256SeedNumWords, kOtbnVarD0, d0));
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256SeedNumWords, kOtbnVarD1, d1));
+}
+
+/**
+ * Generates a keypair from a masked seed.
+ *
+ * The seed shares must be `kEcc256SeedNumWords` words long, and the caller
+ * must provide pre-allocated buffers of the same length for the private key
+ * shares and of length `kEcc256CoordNumWords` for the public key coordinates.
+ *
+ * @param[in] seed  Seed for key generation.
+ * @param[in] mask  Mask for seed.
+ * @param[out] d0   First share of masked private key d.
+ * @param[out] d1   Second share of masked private key d.
+ * @param[out] x    x-coordinate of public key Q.
+ * @param[out] y    y-coordinate of public key Q.
+ */
+static void p256_ecdsa_gen_keypair(const uint32_t *seed, const uint32_t *mask,
+                                   uint32_t *d0, uint32_t *d1, uint32_t *x,
+                                   uint32_t *y) {
+  // Run the key generation program.
+  p256_run_keygen(kEcc256ModeKeypair, seed, mask);
+
+  // Read results.
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256SeedNumWords, kOtbnVarD0, d0));
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256SeedNumWords, kOtbnVarD1, d1));
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256CoordNumWords, kOtbnVarX, x));
+  SS_CHECK_STATUS_OK(otbn_dmem_read(kEcc256CoordNumWords, kOtbnVarY, y));
+}
+
+/**
+ * Simple serial 'k' (secret keygen) command handler.
+ *
+ * Takes the mask value from the simple serial UART and triggers an OTBN
+ * secret key generation operation. The mask must be `kEcc256SeedNumBytes`
+ * bytes long.
+ *
+ * Uses a fixed seed. To overwrite the seed, use the simpleserial command 's'.
+ *
+ * @param[in] mask The mask provided by the simpleserial UART.
+ * @param[in] mask_len Length of the mask.
+ */
+static void ecc256_ecdsa_secret_keygen(const uint8_t *mask, size_t mask_len) {
+  if (mask_len != kEcc256SeedNumBytes) {
+    LOG_ERROR("Invalid mask length %hu", (uint8_t)mask_len);
+    return;
+  }
+
+  // Copy mask to an aligned buffer.
+  uint32_t ecc256_mask[kEcc256SeedNumWords];
+  memcpy(ecc256_mask, mask, kEcc256SeedNumBytes);
+
+  uint32_t ecc256_d0[kEcc256SeedNumWords];
+  uint32_t ecc256_d1[kEcc256SeedNumWords];
+  p256_ecdsa_gen_secret_key(ecc256_seed, ecc256_mask, ecc256_d0, ecc256_d1);
+
+  simple_serial_send_packet('r', (unsigned char *) ecc256_d0, kEcc256SeedNumBytes);
+  simple_serial_send_packet('r', (unsigned char *) ecc256_d1, kEcc256SeedNumBytes);
```

Comment:
I am getting transmission/synchronization errors (second share is `none`) here for `f_clk >91MHz`.
However, I think, we should address this in a different PR, as it requires some discussions on how to deal with it.